src/gui/TextEditor.cpp

```diff
@@ -183,6 +183,12 @@ void TextEditor::iMPreeditChangedCallback(GtkIMContext* context, TextEditor* te)
      */
     gtk_im_context_get_preedit_string(context, &str, &attrs, &cursor_pos);
 
+    if (attrs == nullptr) {
+        attrs = pango_attr_list_new();
+    }
+    te->preeditAttrList = attrs;
```

Comment:
I just realized that we need to unref the previous attribute list.
```suggestion
    if (te->preeditAttrList) {
        pango_attr_list_unref(te->preeditAttrList);
    }
    te->preeditAttrList = attrs;
```